src/storage/mod.rs

```diff
@@ -1264,15 +1263,8 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
                     .get(priority_tag)
                     .inc();
 
-                Self::check_api_version_ranges(
-                    api_version,
-                    ctx.api_version,
-                    CMD,
-                    [(
-                        start_key.as_ref().map(Key::as_encoded),
-                        end_key.as_ref().map(Key::as_encoded),
-                    )],
-                )?;
+                // Remove the check_api_version_ranges,
+                // because of resolve locks range from TiDB is invalid key range mode
```

Comment:
Suggestion:
Do not `check_api_version` in `scan_lock`, to be compatible with TiDB gc-worker,
which resolves locks on regions, and boundary of regions will be out of range of TiDB keys.